### DISCUSSION TITLE
Gateway - wildfly/vert.x

### DISCUSSION THREAD

**Discussion by tsilaRaf** [apiman/apiman#2500] [General]
Hello,

We use actually apiman V2.3.0 ( Gateway wildly).We'd like to migrate to V3.1.x.
In the website of apiman.io,we saw this recommandation:

**The Apiman gateway is distributed in two main versions:
    Vert.x: Our premier and recommended gateway platform.
    WAR: Runnable on various Servlet platforms.**

Could you give us a comparaison of wildfly and vert.x please,

Have you got some feedback on these two technologie?

Thank you,

**Comment by msavy** [comment]
Good morning. 

Vert.x is faster, has lower latency, and lower overall overheads. It uses async throughout the platform, which is beneficial for IO heavy operations that API Gateways are known for.

WAR is something like WildFly or Tomcat. It's easy and familiar, but will have lower performance. For many, the performance difference won't matter.

**Reply by tsilaRaf** [reply]
Thank you for your answer,it's very helpful
